src/spartan/upsnark.rs

```diff
@@ -295,38 +282,43 @@ impl<G: Group, EE: EvaluationEngineTrait<G>> RelaxedR1CSSNARKTrait<G> for R1CSSN
 
       let N_STEPS = pk.num_steps;
 
-      // With uniformity, each entry of the RLC of A, B, C can be expressed using 
+      // With uniformity, each entry of the RLC of A, B, C can be expressed using
       // the RLC of the small_A, small_B, small_C matrices.
 
-      // 1. Evaluate \tilde smallM(r_x, y) for all y 
-      let compute_eval_table_sparse_single= |small_M: &Vec<(usize, usize, G::Scalar)>| -> Vec<G::Scalar> {
-        let mut small_M_evals = vec![G::Scalar::ZERO; pk.S.num_vars + 1];
-        for (row, col, val) in small_M.iter() {
-          small_M_evals[*col] += eq_rx_con[*row] * val;
-        }
-        small_M_evals
-      };
+      // 1. Evaluate \tilde smallM(r_x, y) for all y
+      let compute_eval_table_sparse_single =
+        |small_M: &Vec<(usize, usize, G::Scalar)>| -> Vec<G::Scalar> {
+          let mut small_M_evals = vec![G::Scalar::ZERO; pk.S.num_vars + 1];
+          for (row, col, val) in small_M.iter() {
+            small_M_evals[*col] += eq_rx_con[*row] * val;
+          }
+          small_M_evals
+        };
 
       let (small_A_evals, (small_B_evals, small_C_evals)) = rayon::join(
         || compute_eval_table_sparse_single(&pk.S.A),
-        || rayon::join(
+        || {
+          rayon::join(
```

Comment:
These `rayon::join` trains are gross. We should write a macro to handle these of arbitrary size. Not necessary for this PR.